File: zt-marqueesample/src/main/java/com/zzt/zt_marqueesample/MainActivity.java
package com.zzt.zt_marqueesample;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatTextView;

import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private String textLang = "开始，这是一个长文本 ， 这是一个长文本 ， 这是一个长文本 ， 这是一个长文本 ，结束";
    private TextView tv_m1, tv_m2;
    private  AutoScrollTextView tv_m3 ;
    private  MarqueeHorizontalTextView tv_m4;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        tv_m1 = findViewById(R.id.tv_m1);
        tv_m2 = findViewById(R.id.tv_m2);
        tv_m3 = findViewById(R.id.tv_m3);
        tv_m4 = findViewById(R.id.tv_m4);

        tv_m1.setText(textLang);
        tv_m1.setSelected(true);
        tv_m2.setText(textLang);
        tv_m3.setText(textLang);
        tv_m3.init(getWindowManager());
        tv_m3.startScroll();
        tv_m4.setText(textLang , TextView.BufferType.NORMAL);
    }
}

File: SampleFinger/src/main/java/com/zzt/samplefinger/MainActivity.kt
//package com.zzt.samplefinger
//
//import androidx.appcompat.app.AppCompatActivity
//import android.os.Bundle
//import android.widget.Button
//import androidx.core.hardware.fingerprint.FingerprintManagerCompat
//
//class MainActivity : AppCompatActivity() {
//
//    /**
//     * A class that coordinates access to the fingerprint hardware.
//    On platforms before Build.VERSION_CODES.M, this class behaves as there would be no fingerprint hardware available.
//    Deprecated
//    Use androidx.biometrics.BiometricPrompt instead.
//     */
//    var fingerManger: FingerprintManagerCompat? = null
//
//    var biometricPrompt: BiometricPrompt? = null
//
//    var button: Button? = null
//
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
//        initView()
//    }
//
//    private fun initView() {
//        button = findViewById(R.id.button)
//        button?.setOnClickListener {
//            val biometricManager = BiometricManager.from(this)
//            when (biometricManager.canAuthenticate(BIOMETRIC_STRONG or DEVICE_CREDENTIAL)) {
//                BiometricManager.BIOMETRIC_SUCCESS ->
//                    Log.d("MY_APP_TAG", "App can authenticate using biometrics.")
//                BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE ->
//                    Log.e("MY_APP_TAG", "No biometric features available on this device.")
//                BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE ->
//                    Log.e("MY_APP_TAG", "Biometric features are currently unavailable.")
//                BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED -> {
//                    // Prompts the user to create credentials that your app accepts.
//                    val enrollIntent = Intent(Settings.ACTION_BIOMETRIC_ENROLL).apply {
//                        putExtra(Settings.EXTRA_BIOMETRIC_AUTHENTICATORS_ALLOWED,
//                            BIOMETRIC_STRONG or DEVICE_CREDENTIAL)
//                    }
//                    startActivityForResult(enrollIntent, REQUEST_CODE)
//                }
//            }
//        }
//    }
//
//}